src/rgw/driver/rados/rgw_bucketlogging.cc

```diff
@@ -0,0 +1,605 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "rgw_bucket_logging.h"
+//FIXFIX: better names
+#include "rgw_bucketlogging.h"
+#include "cls/lock/cls_lock_client.h"
+#include <memory>
+#include <boost/algorithm/hex.hpp>
+#include <boost/asio/basic_waitable_timer.hpp>
+#include <boost/asio/executor_work_guard.hpp>
+#include <boost/asio/io_context.hpp>
+#include <boost/asio/spawn.hpp>
+#include <boost/context/protected_fixedsize_stack.hpp>
+#include "include/function2.hpp"
+#include "rgw_sal_rados.h"
+#include "rgw_zone_features.h"
+#include "rgw_perf_counters.h"
+#include "services/svc_zone.h"
+#include "common/dout.h"
+#include "rgw_url.h"
+#include <chrono>
+#include <fmt/format.h>
+#include "librados/AioCompletionImpl.h"
+#include "common/async/yield_waiter.h"
+#include <future>
+#include <thread>
+
+#define dout_subsys ceph_subsys_rgw_bucket_logging
```

Comment:
do we use this in the other bucket logging code or just the manager?
(can be added in subsequent PR)